src/Umbraco.Web/PublishedCache/ContentCacheContentRouter.cs

```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web.PublishedCache.NuCache;
+using CacheKeys = Umbraco.Web.PublishedCache.NuCache.CacheKeys;
+
+namespace Umbraco.Web.PublishedCache
+{
+    internal class ContentCacheContentRouter : IContentRouter
+    {
+        private readonly IPublishedSnapshotAccessor _snapshot;
+        private readonly IGlobalSettings _globalSettings;
+        private readonly IAppCache _snapshotCache;
+        private readonly IAppCache _elementsCache;
+
+        public ContentCacheContentRouter(IPublishedSnapshotAccessor snapshot, IGlobalSettings globalSettings,
```

Comment:
I'm don't think this works as intended. The previous usage of the snapshotCache and elementsCache are not singleton instances and are not injected. The instances of snapshotCache and elementsCache are managed via the PublishedSnapshotService. The PublishedSnapshotService creates a new snapshot cache instance for each generation it creates. This is sort of like a 3 dimensional dictionary and each snapshot/generation is a new layer so that any existing requests continue to use the existing snapshot data that it's operating on. These snapshots are cleaned up when they are no longer in use. Whenever a call to PublishedSnapshotService.GetElements is made, a new snapshotCache instance is used and the existing elementsCache instance is used if a new generation is not needed. In this case these caches will be singletons that are never cleared or updated and will clash between generations. This object should probably not be responsible for any of this caching and the caching responsibility should be where it was before. Also note that `IPublishedSnapshotAccessor` is a request based object that is based on the UmbracoContext, if there is no request, then it will be null so it might not be ideal for this to be based on that. 

I've added notes to this interface declaration as well. I think that for this to work each method would need to take an isntance of `ContentStore.Snapshot` to work on and it would only deal with lookups, no caching. Or something along those lines.

Reply:
I removed the caching, but it will still need to be added back somewhere.

Reply:
Implemented caching